server/controllers/customerController.js

```diff
@@ -0,0 +1,39 @@
+const { Retailer, Customer, Reservation } = require("../models");
+
+const customerController = {
+  getAllRetailers(req, res) {
+    console.log("get all retail route works");
+    Retailer.find(
+      {},
+      "retailerId retailerName address phoneNumber storeHours capacity timers"
+    ).then((results) => {
+      if (!results)
+        res.status(404).json({ message: "No retailers found in database" });
+
+      res.json(results);
+    });
+  },
+  searchRetailers({ params, body }, res) {
+    Retailer.find({ retailerName: new RegExp(params.searchTerm, "i") }).then(
+      (results) => {
+        if (!results || results.length === 0)
+          res
+            .status(404)
+            .json({ message: "No retailers match that search term." });
+        res.json(results);
+      }
+    );
+  },
+  getCustomerReservations({ params }, res) {
+    Customer.find(
+      { customerId: params.customerId },
+      "retailerId queueStatus replyStatus partySize"
```

Comment:
I think this only needs to return "reservations".

Reply:
I mean that I think the only thing needed on line 30 is "reservations".  Please change `Reservation` back to `Customer` and change " retailerId queueStatus replyStatus partySize" to "reservations" (be sure to populate the reservations with all of the referenced documents).  I think the query into the `Customer` collection will be faster than if we query the whole `Reservation` collection.